src/plots/cartesian/axis_defaults.js

```diff
@@ -64,6 +65,10 @@ module.exports = function handleAxisDefaults(containerIn, containerOut, coerce,
         }
     }
 
+    containerOut._initialCategories = axType === 'category' ?
+        orderedCategories(letter, containerIn.categorymode, containerIn.categorylist, options.data) :
```

Comment:
> My original take was that categorymode, as a switch, drives the show in an unambiguous way, and the user can flip-flop among modes without worrying about removing and readding (or commenting/uncommenting) the categorylist property.

... and you would be correct.

What *smart coercion* is trying to remove redundancy. For example,

```js
var layout: {
  xaxis: {
    categorylist: ['apples', 'bananas', 'clementines']
  }
};
```

should be considered the same as:

```js
var layout: {
  xaxis: {
    categorymode: 'array',  // redundant !!!
    categorylist: ['apples', 'bananas', 'clementines']
  }
};
```

But, 

```js
var layout: {
  xaxis: {
    categorymode: 'category descending',  // still used as a switch 
    categorylist: ['apples', 'bananas', 'clementines']  // not used in graph
  }
};
```